code/BuddyTrackerBasic/src/main.cpp

```diff
@@ -99,25 +102,28 @@ void sendPacket(BT_Packet packet){
 // currently adds unknown buddies
 // shows everyone
 void updateBuddy(uint64_t UUID, uint16_t lat_partial, uint16_t lng_partial){
-    uint8_t index = findBuddyIndex(UUID);
+    uint8_t index = findBuddyBy(UUID);
     
     // add unknown buddies
     if(index == -1){
-        Buddy *newBuddy = new Buddy(UUID);
-        buddies.add(newBuddy);
-        index = buddies.size() - 1;
+        if (buddies.size() < MAX_UINT8) {
+          Buddy *newBuddy = new Buddy(UUID);
+          buddies.add(newBuddy);
+          index = buddies.size() - 1;
+          // TODO might be better to get index using findBuddyBy
+        }
     }
 
     // can't compute other Buddy's location without knowning own location
     if(myLat == LAT_LNG_ERR || myLng == LAT_LNG_ERR){
         return;
     }
-    // clear LSBs
-    uint32_t lat = myLat & 0x00;
-    uint32_t lng = myLng & 0x00;
+    // clear 8 LSBs
+    uint32_t lat = myLat & 0xFFFF0000;
```

Comment:
I thought the partial Lat/Long is 16 bits. Make sure to keep upper 16 bits and clear lower 16 bits